Add GitHub Actions release workflow and update GoReleaser config

shreyashx16 opened the pull request:
# chore(ci): Add release workflow and update GoReleaser config

This PR adds a GitHub Actions workflow to automate the release process using GoReleaser. It builds the binaries for Linux and macOS, creates GitHub Releases with artifacts, and verifies the release binaries by running some basic checks.


### What’s included in this PR:

- `.github/workflows/release.yml`: The GitHub Actions workflow that runs on tag pushes (`v*`). It:
  - Checks out the code.
  - Sets up Go environment.
  - Builds binaries for Linux and macOS.
  - Uploads build artifacts.
  - Runs GoReleaser to create GitHub Releases.
  - Downloads the released assets and verifies their functionality.
  - Prepares and deploys an APT repository for `.deb` packages.
  
- `.goreleaser.yml`: GoReleaser configuration updated with:
  - Build targets for Linux and macOS (amd64 and arm64).
  - Release settings including pre-release detection.
  - Packaging configurations for `.deb` and optional Homebrew (commented out by default).


### Notes for the Maintainer:

To enable the release workflow fully, a **GitHub Personal Access Token (PAT)** needs to be created and added as a secret named `GH_PAT` in the repository’s Settings → Secrets.

**Required permissions for the token:**

- `repo` scope — to create and manage releases and upload release assets.
- `workflow` scope — to trigger and manage workflows if needed.

Optionally, to enable Homebrew publishing (currently commented out for safety), the maintainer can create another token with the `repo` scope and add it as `HOMEBREW_GITHUB_TOKEN`.


### Why are some parts commented out?

The Homebrew publishing section is commented out to prevent accidental pushes to the upstream formula repository from forks. Once this PR is merged upstream and the maintainer is ready, they can safely uncomment and configure the Homebrew publishing with the appropriate token.


### Testing:

All steps have been tested locally, including building, packaging, and verifying the binaries. The workflow is designed to trigger only on tag pushes following the pattern `v*` to prevent unintended releases.

### Screenshots 

<img width="1912" height="834" alt="Screenshot 2025-10-11 025639" src="https://github.com/user-attachments/assets/e6bcbb73-9fbe-4841-940c-ef163d4ed293" />

<img width="1707" height="874" alt="Screenshot 2025-10-11 023519" src="https://github.com/user-attachments/assets/bbb0f01b-eb14-466d-8b54-6e4431d8a878" />

Fixes
#4 


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Added macOS and Linux release binaries with platform-specific artifacts and published Debian packages in a versioned APT layout.

- Tests
  - Automated verification of released binaries across OS/architectures, including extraction and version checks and optional .deb install validation.

- Chores
  - CI moved to matrix builds, updated action versions, improved artifact naming/upload, hardened release/publish steps, and ignored build artifacts.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

shreyashx16 committed:
chore(ci): add release workflow and goreleaser config

shreyashx16 committed:
chore: update .gitignore

Agent-Hellboy commented:
Damn, cool bro, let me review

Agent-Hellboy commented:
@coderabbitai please review

Agent-Hellboy commented:
If you can, please help me with https://github.com/Agent-Hellboy/mcp-server-fuzzer/issues/100

shreyashx16 committed:
fix(ci): upgrade deprecated GitHub Actions in release.yml to Node 20–compatible versions

Agent-Hellboy commented:
Hi @yashx007 , i am adding a patchset, thanks for your help, please help with the repo i suggest, it has a lot of potential, please spread it among your friends as well,  need results of a lot of open open source servers, they can add a lot of PR as part of hacktoberfest and it's easy agentic editor can help them.

shreyashx16 commented:
@Agent-Hellboy do i need to fix the change suggested by CodeRabbit? or my work is done?

shreyashx16 commented:
> Hi @yashx007 , i am adding a patchset, thanks for your help, please help with the repo i suggest, it has a lot of potential, please spread it among your friends as well, need results of a lot of open open source servers, they can add a lot of PR as part of hacktoberfest and it's easy agentic editor can help them.

I will try that but i dont have time right now maybe in the upcoming week.

Agent-Hellboy commented:
> @Agent-Hellboy do i need to fix the change suggested by CodeRabbit? or my work is done?

No ,  I will do it

Agent-Hellboy commented:
> I will try that but i dont have time right now maybe in the upcoming week.

no problem, spread it among your friends if their PR is pending.

Agent-Hellboy committed:
Update release workflow

Agent-Hellboy commented:
Again @yashx007, I am looking for a contributor for https://github.com/Agent-Hellboy/mcp-server-fuzzer. It has funding as well and has the potential to secure additional funding. If you are in college, please share it with your friends.

shreyashx16 commented:
@Agent-Hellboy None of my friends are into open source, it's just me and I can help you but currently I have placement related work, but I will let u know shortly

Agent-Hellboy committed:
fix: use dynamic GOOS/GOARCH for accurate binary naming

- Replace hardcoded architecture assumptions with dynamic detection
- Fixes macOS arm64 runners being labeled as amd64
- Ensures artifact names match actual binary architecture
